Documentation of Mesh REST API added

torsannes opened the pull request:
First version!
Is the structure OK?
Is something missing?
...

tnoczyns-volue commented on docs/mesh/rest-api/generate-c#-api-from-swagger-json.md:
```diff
@@ -0,0 +1,71 @@
+
+[here]: https://github.com/RicoSuter/NSwag/wiki/NSwagStudio
+
+### Configure a NSwagStudi project
```
```suggestion
### Configure a NSwagStudio project
```

tnoczyns-volue reviewed (commented):
Can we keep the images in `assets/images` like in e.g. mesh/concepts folder.

Please fix REST API related warnings:
```
INFO    -  Cleaning site directory
INFO    -  Building documentation to directory: C:\dev\repos\energy\energy-smp-docs\site
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
             - mesh\calculations\timepoint-macros.md
             - mesh\rest-api\how-to-use-ImportMultistepWaterValueFunctions.md
             - mesh\rest-api\how-to-use-the-mesh-rest-api.md
WARNING -  A reference to 'how-to-use-ImportMultistepWaterValueFunctions.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  Doc file 'mesh-data-transfer/changelog/CHANGELOG.md' contains a link './src/DatabaseGateway/README.md', but the target 'mesh-data-transfer/changelog/src/DatabaseGateway/README.md' is not found among documentation files.
WARNING -  Path 'js\version-select.js' uses OS-specific separator '\'. That will be unsupported in a future release. Please change it to '/'.
WARNING -  Path 'css\version-select.css' uses OS-specific separator '\'. That will be unsupported in a future release. Please change it to '/'.
INFO    -  Doc file 'mesh/rest-api/how-to-use-the-mesh-rest-api.md' contains a link '#using-oauth2-security-token', but there is no such anchor on this page.
INFO    -  Doc file 'mesh/rest-api/how-to-use-the-mesh-rest-api.md' contains a link '#using-kerberos-user-information', but there is no such anchor on this page.
INFO    -  Doc file 'mesh/rest-api/introduction.md' contains a link '#using-oauth2-security-token', but there is no such anchor on this page.
INFO    -  Doc file 'mesh/rest-api/introduction.md' contains a link '#using-kerberos-user-information', but there is no such anchor on this page.
INFO    -  Doc file 'timeseries-reports/sort_contents.md' contains a link '#Main', but there is no such anchor on this page.
INFO    -  Documentation built in 3.48 seconds
```

torsannes commented:
> Can we keep the images in `assets/images` like in e.g. mesh/concepts folder.
> 
> Please fix REST API related warnings:
> 
> ```
> INFO    -  Cleaning site directory
> INFO    -  Building documentation to directory: C:\dev\repos\energy\energy-smp-docs\site
> INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
>              - mesh\calculations\timepoint-macros.md
>              - mesh\rest-api\how-to-use-ImportMultistepWaterValueFunctions.md
>              - mesh\rest-api\how-to-use-the-mesh-rest-api.md
> WARNING -  A reference to 'how-to-use-ImportMultistepWaterValueFunctions.md' is included in the 'nav' configuration, which is not found in the documentation files.
> WARNING -  Doc file 'mesh-data-transfer/changelog/CHANGELOG.md' contains a link './src/DatabaseGateway/README.md', but the target 'mesh-data-transfer/changelog/src/DatabaseGateway/README.md' is not found among documentation files.
> WARNING -  Path 'js\version-select.js' uses OS-specific separator '\'. That will be unsupported in a future release. Please change it to '/'.
> WARNING -  Path 'css\version-select.css' uses OS-specific separator '\'. That will be unsupported in a future release. Please change it to '/'.
> INFO    -  Doc file 'mesh/rest-api/how-to-use-the-mesh-rest-api.md' contains a link '#using-oauth2-security-token', but there is no such anchor on this page.
> INFO    -  Doc file 'mesh/rest-api/how-to-use-the-mesh-rest-api.md' contains a link '#using-kerberos-user-information', but there is no such anchor on this page.
> INFO    -  Doc file 'mesh/rest-api/introduction.md' contains a link '#using-oauth2-security-token', but there is no such anchor on this page.
> INFO    -  Doc file 'mesh/rest-api/introduction.md' contains a link '#using-kerberos-user-information', but there is no such anchor on this page.
> INFO    -  Doc file 'timeseries-reports/sort_contents.md' contains a link '#Main', but there is no such anchor on this page.
> INFO    -  Documentation built in 3.48 seconds
> ```

- pictures directory is renamed images
- fixed path in the 'nav' configuration
- fixed internal references
- removed 'how-to-use-the-mesh-api.md' file (content was already included in the 'introduction.md' file

tnoczyns-volue commented on docs/mesh/rest-api/api-documentation.md:
```diff
@@ -0,0 +1,2757 @@
+
+*The minimal information of a specific Mesh object.*
+
+Type: objectd
```
```suggestion
Type: object
```

??

tnoczyns-volue commented on docs/mesh/rest-api/changelog.md:
```diff
@@ -0,0 +1,93 @@
+
+- We have fixed an issue when reading values using resource path or timeseriesKey to identify the time series. This fix involves the following methods: ReadTimeSeriesValues, ReadTimeSeriesValuesRaw, ReadTimeSeriesValuesMulti and ReadTimeSeriesValuesMultiRaw. (#9)
+- We have enhanced the exception messages returned from the API to better pin-point the source of the exception. Exceptions originating from the Mesh service is containing a `MESH - ` part, otherwise it contains an `API - ` part. (#10)
+- We have fixed exceptions from the ReadTimeSeriesValuesMulti, ReadTimeSeriesValuesMultiRaw and WriteTimeSerisValuesMultiRaw methods to always return reference to the time series if the exception is related to a specific time series. (#11)
```
```suggestion
- We have fixed exceptions from the ReadTimeSeriesValuesMulti, ReadTimeSeriesValuesMultiRaw and WriteTimeSeriesValuesMultiRaw methods to always return reference to the time series if the exception is related to a specific time series. (#11)
```

tnoczyns-volue commented on mkdocs.yml:
```diff
@@ -157,6 +157,13 @@ nav:
+          - Mesh REST API:
+                - Introduction: mesh/rest-api/introduction.md
+                - Changelog: mesh/rest-api/changelog.md
+                - API documentation: mesh/rest-api/api-documentation.md
```
```suggestion
                - API specification: mesh/rest-api/api-documentation.md
```

To be consistent with gRPC section.

tnoczyns-volue reviewed (approved):
Just nitpicks.
